Plugins/Json/src/ProtoAxisJsonConverter.cpp

```diff
@@ -0,0 +1,56 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/Json/ProtoAxisJsonConverter.hpp"
+
+#include "Acts/Plugins/Json/GridJsonConverter.hpp"
+#include "Acts/Plugins/Json/UtilitiesJsonConverter.hpp"
+#include "Acts/Utilities/AxisDefinitions.hpp"
+
+nlohmann::json Acts::ProtoAxisJsonConverter::toJson(const Acts::ProtoAxis& pa) {
+  nlohmann::json j;
+  j["axis_dir"] = pa.getAxisDirection();
+  j["axis"] = AxisJsonConverter::toJson(pa.getAxis());
+  j["autorange"] = pa.isAutorange();
+  return j;
+}
+
+Acts::ProtoAxis Acts::ProtoAxisJsonConverter::fromJson(
+    const nlohmann::json& j) {
+  auto axisDir = j.at("axis_dir").get<Acts::AxisDirection>();
+  auto axisBoundaryType =
+      j.at("axis").at("boundary_type").get<Acts::AxisBoundaryType>();
+  if (auto axisType = j.at("axis").at("type").get<Acts::AxisType>();
+      axisType == AxisType::Equidistant) {
+    if (j.at("autorange").get<bool>()) {
+      auto nbins = j.at("axis").at("bins").get<std::size_t>();
+      if (nbins == 0) {
+        throw std::invalid_argument("Number of bins must be positive");
+      }
+      return ProtoAxis(axisDir, axisBoundaryType, nbins);
+    }
+    auto min = j.at("axis").at("range").at(0).get<double>();
+    auto max = j.at("axis").at("range").at(1).get<double>();
+    auto nbins = j.at("axis").at("bins").get<std::size_t>();
+    if (min >= max) {
+      throw std::invalid_argument("Invalid range: min must be less than max");
```

Comment:
Should we also add tests for this and the other exceptions?

Reply:
Let me see, I will have to make a mockup `json` for this, but I think that's a good idea.